lib/Db/VaultMapper.php

```diff
@@ -24,61 +24,86 @@
 namespace OCA\Passman\Db;
 
 use OCA\Passman\Utility\Utils;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Db\Entity;
+use OCP\AppFramework\Db\MultipleObjectsReturnedException;
+use OCP\AppFramework\Db\QBMapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
 use OCP\IDBConnection;
-use OCP\AppFramework\Db\Mapper;
 
-class VaultMapper extends Mapper {
-	private $utils;
+class VaultMapper extends QBMapper {
+	const TABLE_NAME = 'passman_vaults';
+	private Utils $utils;
+
 	public function __construct(IDBConnection $db, Utils $utils) {
-		parent::__construct($db, 'passman_vaults');
+		parent::__construct($db, self::TABLE_NAME);
 		$this->utils = $utils;
 	}
 
 
 	/**
-	 * @throws \OCP\AppFramework\Db\DoesNotExistException if not found
-	 * @throws \OCP\AppFramework\Db\MultipleObjectsReturnedException if more than one result
+	 * @param int $vault_id
+	 * @param string $user_id
 	 * @return Vault[]
 	 */
-	public function find($vault_id, $user_id) {
-		$sql = 'SELECT * FROM `*PREFIX*passman_vaults` ' .
-			'WHERE `id`= ? and `user_id` = ?';
-		return $this->findEntities($sql, [$vault_id, $user_id]);
+	public function find(int $vault_id, string $user_id) {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('*')
+			->from(self::TABLE_NAME)
+			->where($qb->expr()->eq('id', $qb->createNamedParameter($vault_id, IQueryBuilder::PARAM_INT)))
+			->andWhere($qb->expr()->eq('user_id', $qb->createNamedParameter($user_id, IQueryBuilder::PARAM_STR)));
+
+		/** @var Vault[] $vaults */
+		$vaults = $this->findEntities($qb);
+		return $vaults;
 	}
+
 	/**
-	 * @throws \OCP\AppFramework\Db\DoesNotExistException if not found
-	 * @throws \OCP\AppFramework\Db\MultipleObjectsReturnedException if more than one result
+	 * @param string $vault_guid
+	 * @param string $user_id
 	 * @return Vault
+	 * @throws DoesNotExistException
+	 * @throws MultipleObjectsReturnedException
 	 */
-	public function findByGuid($vault_guid, $user_id) {
-		$sql = 'SELECT * FROM `*PREFIX*passman_vaults` ' .
-			'WHERE `guid`= ? and `user_id` = ?';
-		return $this->findEntity($sql, [$vault_guid, $user_id]);
+	public function findByGuid(string $vault_guid, string $user_id) {
+		$qb = $this->db->getQueryBuilder();
+		$qb->select('*')
+			->from(self::TABLE_NAME)
+			->where($qb->expr()->eq('guid', $qb->createNamedParameter($vault_guid, IQueryBuilder::PARAM_STR)))
+			->andWhere($qb->expr()->eq('user_id', $qb->createNamedParameter($user_id, IQueryBuilder::PARAM_STR)));
+
+		/** @var Vault $vault */
+		$vault = $this->findEntity($qb);
```

Comment:
This var is never used

Reply:
were IDE optimizations, they are all removed again. Something with "clean code" :)